Add Vertcoin support

metalicjames opened the pull request:
This PR implements the `vtcatomicswap` command.

jrick commented:
Thank you for contributing this!

It's late here and I can't do a thorough review of this right now, but will make a few quick remarks.

First we prefer to have linear history, so if you could rebase and squash this work over master, that would be appreciated.

Second I don't see any of the vertcoin deps added to the dep files (Gopkg.toml, and after running dep ensure, Gopkg.lock).  This will need to happen since we require vendoring our dependencies for stability and security.

Third, you should add yourself to the copyright.  Don't short change yourself!  I think it may also make sense to move the LICENSE file in the root dir to each individual project so it can be updated individually.  I can take care of that in a separate PR.

Finally I'd like to test this with a swap on testnet.  Would you be willing to help test with this?  I can begin syncing a vertcoin testnet chain tonight.

jrick commented:
Testnet may have to wait a bit, all i see in debug.log is this message repeatedly:

```
2017-09-23 05:23:58 connect() to 5.196.74.100:15889 failed after select(): Connection refused (111)
```

and it doesn't appear to be trying any other nodes:

```
2017-09-23 05:23:58 1 addresses found from DNS seeds
```

metalicjames commented:
Okay, I will make those changes and get them back to you (I was just building with `go build`). As for testnet, no one really uses it much, let me put up a testnet node for you to connect to: fr1.vtconline.org. It may be easier for us to test a small amount on main net.

jrick commented on cmd/vtcatomicswap/main.go:
```diff
@@ -0,0 +1,1010 @@
+		return nil, err
+	}
+
+	unsignedContract := wire.NewMsgTx(2)
```
Should this be version 1 or 2?  Looking through vertcoin's transaction.h, version 1 is the current version while 2 is the max standard version, implying that version 2 transactions are't ready to send yet.

https://github.com/vertcoin/vertcoin/blob/ec464a06f35da791c829b6f745b35d2ed9ca5a8b/src/primitives/transaction.h#L306

metalicjames replied:
We've been using version 2 transactions for LN and SegWit. If/when we release a new Core update that can send to/receive such transactions natively we'll bump the default transaction version there too (e.g. https://vtc.xchan.gr/blockExplorer/tx/3d0ba63444e75c7d751a8f2ba081e44cac60d0cabb2c0bb1bfa7ec1bfa40b717).

I assume using version 1 there would be fine then given that this code doesn't use segwit.

jrick commented:
Just to keep you updated on recent developments, a sixth command `refund` was added to the tools.  vtcatomicswap will need to support it too (should be a straight port from the btc/ltc tools).

metalicjames commented:
Rebased to the latest code from master. I _think_ I successfully updated the Gopkg.lock file and I updated the copyright notices. This should be ready to merge after you're satisfied.

jrick commented on README.md:
```diff
@@ -2,9 +2,8 @@
-Core) and Litecoin (Litecoin Core) are the two other blockchains and wallets
-supported.  Support for other blockchains or wallets could be added in the
-future.
+Core), Litecoin (Litecoin Core) and Vertcoin (Vertcoin Core) are the three other blockchains and wallets supported.  Support for other blockchains or wallets 
```
please keep this word wrapped to 80 columns

jrick commented on cmd/vtcatomicswap/main.go:
```diff
@@ -0,0 +1,1106 @@
+// Copyright (c) 2017 The Decred developers.
+//				2017 The Vertcoin developers.
```
Weird spacing here and there should be two copyright notices:

```
// Copyright (c) 2017 The Decred developers.
// Copyright (c) 2017 The Vertcoin developers.
```

jrick commented on cmd/vtcatomicswap/sizeest.go:
```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2016 The btcsuite developers
+//               2016-2017 The Decred developers
+//               2017 The Vertcoin developers
```
Same idea here:

```
// Copyright (c) 2016 The btcsuite developers
// Copyright (c) 2016-2017 The Decred developers
// Copyright (c) 2017 The Vertcoin developers
```

metalicjames commented:
Okay, those should all be fixed now. I didn't need to make any changes to the `Gopkg.toml` file (should I have done?).

jrick commented:
The digest of the Gopkg.toml is different (this is recorded in the Gopkg.lock to help prevent adding or removing constraints without updating both), and we want to include all of the vertcoin deps in the Gopkg.toml, otherwise if anyone updates, the vertcoin deps will no longer be there.

```
[jrick@traveler atomicswap]$ rm -rf vendor
[jrick@traveler atomicswap]$ dep ensure -update
[jrick@traveler atomicswap]$ go install ./cmd/vtcatomicswap/
cmd/vtcatomicswap/main.go:24:2: cannot find package "github.com/vertcoin/vtcd/chaincfg" in any of:
...
```

metalicjames commented:
Okay, updated now.
